packages/cli/src/controllers/users.controller.ts

```diff
@@ -354,20 +354,64 @@ export class UsersController {
 			was_disabled_ldap_user: false,
 		});
 
-		await this.externalHooks.run('user.profile.update', [invitee.email, sanitizeUser(invitee)]);
+		const publicInvitee = await this.userService.toPublic(invitee);
+
+		await this.externalHooks.run('user.profile.update', [invitee.email, publicInvitee]);
 		await this.externalHooks.run('user.password.update', [invitee.email, invitee.password]);
 
-		return withFeatureFlags(this.postHog, sanitizeUser(updatedUser));
+		return this.userService.toPublic(updatedUser, { posthog: this.postHog });
 	}
 
 	@Authorized('any')
-	@Get('/')
-	async listUsers(req: UserRequest.List) {
-		const users = await this.userService.findMany({ relations: ['globalRole', 'authIdentities'] });
-		return users.map(
-			(user): PublicUser =>
-				addInviteLinkToUser(sanitizeUser(user, ['personalizationAnswers']), req.user.id),
+	@Get('/', { middlewares: listQueryMiddleware })
+	async listUsers(req: ListQuery.Request) {
+		const { listQueryOptions } = req;
+
+		const findManyOptions: FindManyOptions<User> = {};
+
+		if (!listQueryOptions) {
+			findManyOptions.relations = ['globalRole', 'authIdentities'];
+		} else {
+			const { filter, select, take, skip } = listQueryOptions;
+
+			if (filter) findManyOptions.where = filter;
+			if (select) findManyOptions.select = select;
+			if (take) findManyOptions.take = take;
+			if (skip) findManyOptions.skip = skip;
+
+			if (filter?.isOwner !== undefined) {
+				findManyOptions.relations = ['globalRole'];
```

Comment:
One consequence of this line is that `select` always brings the `globalRole` regardless of the fields you choose.

Reply:
A bit more detail here? The condition only adds `globalRole` if we're filtering by `isOwner`.

`select` with `firstName` only brings that field:

<img width="989" alt="Capture 2023-08-25 at 12 02 23@2x" src="https://github.com/n8n-io/n8n/assets/44588767/747ccb77-6378-4a21-9ad6-781838ef55bb">


`filter` with `isOwner` only brings the owner:

<img width="1051" alt="Capture 2023-08-25 at 12 03 30@2x" src="https://github.com/n8n-io/n8n/assets/44588767/be2475e4-7228-415c-9ad4-507f32dfd13e">

Reply:
Ah you're right, I was filtering by `isOwner`. All good.